core/unit_test/TestRangePolicyConstructors.hpp

```diff
@@ -158,6 +166,15 @@ TEST(TEST_CATEGORY_DEATH, range_policy_implicitly_converted_bounds) {
     ASSERT_DEATH({ (void)UIntPolicy(test_val, 10, Kokkos::ChunkSize(2)); },
                  get_error_msg(expected, test_val));
   }
+  {
+    using test_type = TestIndexConversionCheck<typename IntPolicy::index_type>;
+
+    static_assert(std::is_convertible_v<test_type, IntPolicy::index_type>);
+    static_assert(!std::is_convertible_v<IntPolicy::index_type, test_type>);
+
+    typename IntIndexType::type test_val = -1;
+    ASSERT_NO_THROW((void)IntPolicy(-2, test_type(&test_val)));
```

Comment:
```suggestion
    using test_type = TestIndexConversionCheck<UIntPolicy::index_type>;

    static_assert(std::is_convertible_v<test_type, UIntPolicy::index_type>);
    static_assert(!std::is_convertible_v<UIntPolicy::index_type, test_type>);

    UIntPolicy::index_type test_val = -1;
    ASSERT_NO_THROW((void)UIntPolicy(-2, &test_val));
```
When you pick the signed integer version, you only test that it compiles.  If you pick the unsigned one, you also reveal that we would not currently catch that the issue.  I think it makes it a better choice.

Reply:
Converted to unsigned.